bitnami/mediawiki/1/debian-12/rootfs/opt/bitnami/scripts/libmediawiki.sh

```diff
@@ -317,14 +322,14 @@ mediawiki_configure_host() {
         url="https://${host}"
         [[ "$MEDIAWIKI_EXTERNAL_HTTPS_PORT_NUMBER" != "443" ]] && url+=":${MEDIAWIKI_EXTERNAL_HTTPS_PORT_NUMBER}"
     else
-        if [[ "$MEDIAWIKI_EXTERNAL_HTTP_PORT_NUMBER" != "80" || "$MEDIAWIKI_EXTERNAL_HTTPS_PORT_NUMBER" != "443" ]]; then
-            url="http://${host}"
-            [[ "$MEDIAWIKI_EXTERNAL_HTTP_PORT_NUMBER" != "80" ]] && url+=":${MEDIAWIKI_EXTERNAL_HTTP_PORT_NUMBER}"
-        else
-            # If using default values, support both HTTP and HTTPS at the same time
-            url="//${host}"
-        fi
+      url="http://${host}"
+      [[ "$MEDIAWIKI_EXTERNAL_HTTP_PORT_NUMBER" != "80" ]] && url+=":${MEDIAWIKI_EXTERNAL_HTTP_PORT_NUMBER}"
     fi
-    mediawiki_conf_set "\$wgServer" "$url"
 
+    mediawiki_conf_set "\$wgServer" ""
+    # these keys aren't included in the default LocalSettings
+    cat >> "$MEDIAWIKI_CONF_FILE" <<EOF
+\$wgCanonicalServer = "$url";
+\$wgEnableCanonicalServerLink = true;
+EOF
```

Comment:
Could you please provide more context about these changes? Why setting `$wgServer` to an empty string and removing the protocol-relative conditional? (`url="//${host}"`)

According to [mediawiki documentation](https://www.mediawiki.org/wiki/Manual:$wgServer):
> Since version 1.18.0 MediaWiki also supports setting $wgServer to a protocol-relative URL (e.g., //www.mediawiki.org). This is used for supporting both HTTP and HTTPS with the same caches by using links that work under both protocols. When doing this, [$wgCanonicalServer](https://www.mediawiki.org/wiki/Special:MyLanguage/Manual:$wgCanonicalServer) can be used to set the full URL including protocol that will be used in locations such as emails that do not support protocol relative URLs. If the URL starts with https://, MediaWiki will assume that your wiki prefers the HTTPS protocol or supports it exclusively. In that case, you should also set [$wgForceHTTPS](https://www.mediawiki.org/wiki/Special:MyLanguage/Manual:$wgForceHTTPS). See [Manual:HTTPS](https://www.mediawiki.org/wiki/Special:MyLanguage/Manual:HTTPS) for more information.

Reply:
i'm setting `$wgServer` to an empty string since it's used by MediaWiki to generate links, which results in absolute URLs in the application interface instead of relative ones.  this adds needless complexity when configuring a reverse proxy, or just when running locally in development.

Reply:
Understood, thanks you for your comment!